fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/service/PaginationHelper.java

```diff
@@ -41,18 +43,32 @@ public PaginationHelper(DatabaseSpecificSQLGenerator sqlGenerator, DatabaseTypeR
 
     public <E> Page<E> fetchPage(final JdbcTemplate jt, final String sqlFetchRows, final Object[] args, final RowMapper<E> rowMapper) {
 
-        final List<E> items = jt.query(sqlFetchRows, rowMapper, args);
+        final List<E> items = jt.query(connection -> {
+            final PreparedStatement preparedStatement = connection.prepareStatement(sqlFetchRows);
```

Comment:
Why do we need to construct the prepared statement manually? We could use the jdbcTemplate's query method with pure SQL strings. Internally that'll be turned into prepared statements.